api-helpers/contentCreatorFunctions.ts

```diff
@@ -0,0 +1,152 @@
+import { resolve } from 'path';
+import { URL } from 'url';
+
+import Boom from '@hapi/boom';
+import cheerio from 'cheerio';
+import Slugify from 'slugify';
+
+import { closeConnection, openConnection } from './db';
+
+const NEVER = new Date(0);
+
+type BlogData = {
+  readonly name: string;
+  readonly href: string;
+  readonly favicon: string;
+  readonly feeds: readonly Feed[];
+};
+
+type Feed = {
+  readonly type: string;
+  readonly url: string;
+  readonly title: string;
+};
+
+export const addContentCreator = async (url: string, email?: string) => {
+  const prisma = await openConnection();
+  try {
+    const { name, href, favicon, feeds } = await getBlogData(url);
+    return prisma.blog.create({
+      data: {
+        name,
+        href,
+        favicon,
+        rss: feeds[0].url, // @to-do Ustalić, który jest najlepszy
+        slug: Slugify(name, { lower: true }),
+        lastUpdateDate: NEVER,
+      },
+    });
+  } finally {
+    await closeConnection();
+  }
+};
+
+const getBlogData = async (url: string): Promise<BlogData> => {
+  const youtubeRss = getYoutubeRss(url);
+  if (youtubeRss) {
+    return {
+      name: 'Some Youtube Channel', // @to-do Ustalić, nazwę strony z yt
+      href: url,
+      favicon: 'https://www.youtube.com/s/desktop/d743f786/img/favicon.ico',
+      feeds: [
+        {
+          type: '',
+          url: youtubeRss,
+          title: youtubeRss,
+        },
+      ],
+    };
+  } else {
+    const response = await fetch(url);
+    const htmlText = await response.text();
+    if (htmlText) {
+      const $ = cheerio.load(htmlText);
+      const blogName = getBlogName($);
```

Comment:
Branie nazwy bloga z title w html to nie jest dobry pomysł. Przykład? 

![image](https://user-images.githubusercontent.com/24841038/103142585-bc084800-4705-11eb-9707-0416c789953a.png)

Reply:
Zasugeruj lepszą alternatywę

Reply:
Po znalezieniu adresu RSS'a, tytuł można pobrać z RSS'a (tak samo jak nazwę kanału na YT)

Reply:
I tam title będzie lepszy? Pokaż na przykładzie typeofweb.com

Reply:
Tak

![image](https://user-images.githubusercontent.com/24841038/103142885-df35f600-470b-11eb-98d9-6b926a13d37e.png)

Reply:
Okej, przekonałeś mnie.